### PR TITLE
http_server: add http handler for failpoint

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- Lack of a convenient way to control failpoints at runtime.

### What is changed and how it works?
- Added an http API endpoint as `/fail/`

### Check List 

Tests 

 - Unit test
 
Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### Release note

- Aadd http handler for failpoint

### CODE REVIEW

**Review by amyangfei — APPROVED** [review]

**Review by lonng — COMMENTED** [review]
/lgtm

File: cdc/http_status.go [hunk 1]
```diff
@@ -54,6 +55,9 @@ func (s *Server) startStatusHTTP() error {
 
 	serverMux.HandleFunc("/admin/log", handleAdminLogLevel)
 
+	// `http.StripPrefix` is needed because `failpoint.HttpHandler` assumes that it handles the prefix `/`.
+	serverMux.Handle("/fail/", http.StripPrefix("/fail", &failpoint.HttpHandler{}))
```
Comment (leoppro) [inline cdc/http_status.go:59]: check if the failpoint is enable

Comment (JinLingChristopher) [inline cdc/http_status.go:59]: how about change "/fail/" to "/fail", the last '/' is redundant I think.

**Comment by liuzix** [comment]
run-all-tests

File: cdc/http_status.go [hunk 2]
```diff
@@ -54,6 +57,11 @@ func (s *Server) startStatusHTTP() error {
 
 	serverMux.HandleFunc("/admin/log", handleAdminLogLevel)
 
+	if util.FailpointBuild {
+		// `http.StripPrefix` is needed because `failpoint.HttpHandler` assumes that it handles the prefix `/`.
+		serverMux.Handle("/fail/", http.StripPrefix("/fail", &failpoint.HttpHandler{}))
```
Comment (overvenus) [inline cdc/http_status.go:62]:
```suggestion
		serverMux.Handle("/debug/fail/", http.StripPrefix("/debug/fail", &failpoint.HttpHandler{}))
```

I prefer to add a prefix /debug.

**Review by overvenus — COMMENTED** [review]
rest LGTM

**Pull request merged**